boskos/janitor/janitor.py

```diff
@@ -143,11 +144,60 @@ def clear_resources(project, cols, resource):
         try:
             subprocess.check_call(base + list(items))
         except subprocess.CalledProcessError as exc:
-            err = 1
+            if not resource.tolerate:
+                err = 1
             print >>sys.stderr, 'Error try to delete resources: %r' % exc
     return err
 
 
+def clean_gke_cluster(project, age, filt):
+    """Clean up potential leaking gke cluster"""
+
+    # a cluster can be created in one of those three endpoints
+    endpoints = [
+        'https://test-container.sandbox.googleapis.com/', # test
+        'https://staging-container.sandbox.googleapis.com/', # staging
+        'https://container.googleapis.com/', # prod
+    ]
+
+    err = 0
+    for endpoint in endpoints:
+        os.environ['CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER'] = endpoint
+        print "checking endpoint %s" % endpoint
+        cmd = [
+            'gcloud', 'container', '-q', 'clusters', 'list',
+            '--project=%s' % project,
+            '--filter=%s' % filt,
+            '--format=json(name,createTime)'
+            ]
+        print 'running %s' % cmd
+        for item in json.loads(subprocess.check_output(cmd)):
+            print 'cluster info: %r' % item
+            if 'name' not in item or 'createTime' not in item:
+                print >>sys.stderr, 'name and createTime must present'
+                raise ValueError('%r' % item)
+
+            item['createTime'] = item['createTime'].split('+')[0]
+            created = datetime.datetime.strptime(
```

Comment:
is this timezone safe?

Reply:
tried a few of our projects, looks like they are all ends with `+00:00`

Reply:
maybe toss a comment about that here

Reply:
done